### ISSUE TITLE
List: allow indenting with Tab 

### ISSUE THREAD

**Issue by Marc-pi** [WordPress/gutenberg#45404]
## What problem does this address?
It is not possible to ident Lists quickly with tabs keyboard button. 
The only way actually is to use the toolbar button.
It is a major need for all of us that are daily building contents with WP FSE.

- I've checked within the issue tracker in Gutenberg Plugin backlog, 
- I've checked also in Gutenberg 14.4/14.5
- I did not find quickly the List v2 Roadmap, maybe it is in it already ?

Let's poke @ellatrix who seems to manage the block improvements with dexterity ;-)

![image](https://user-images.githubusercontent.com/919332/198875965-6b6042b8-7723-4899-929b-43e1ffa80fdc.png)

## What is your proposed solution?
Just like in Google Docs or MS Word, since users took (for ages) the habit to use the tabs to indent lists.

**Comment by ellatrix** [comment]
We will add Tab. Requires better methods to navigate in the editor though. See https://github.com/WordPress/gutenberg/pull/45157#issuecomment-1291720149.

**Comment by Marc-pi** [comment]
ok, thanks
Needed, just for memo : 
* Tab (indent)
* Shift + Tab (un-indent)

**Comment by danielbachhuber** [comment]
> Needed, just for memo :
> 
>     * Tab (indent)
> 
>     * Shift + Tab (un-indent)

This works great in the Classic Block 😊

Previously https://github.com/WordPress/gutenberg/issues/4181 and https://github.com/WordPress/gutenberg/issues/9611

**Comment by mtias** [comment]
> The only way actually is to use the toolbar button.

You can actually use space and backspace to indent and unindent with the caret on the first position of a list. The only reason tab doesn't work is because there's an accessibility conflict with regards to navigation expectations. The solution to which has been elusive. The most promising path is to couple the behaviors with the "constrain focus" setting.

**Comment by alexstine** [comment]
Yes, this will be possible when Tab/Shift+Tab are no longer reserved navigation keys and they can be block-based actions. Until then, this is blocked.

**Comment by simison** [comment]
Would highlight a list item, and then pressing tab while list item is highlighted work for accessibility in Gutenberg? Similar works in browsers natively.

Screen recording below:

https://github.com/WordPress/gutenberg/assets/87168/8b99b0d1-8a72-4a02-8178-c767ddede168

**Comment by alexstine** [comment]
That is probably fine for A11Y but not very discoverable. Needs docs at the very least.

**Comment by ellatrix** [comment]
@simison I don't think that's native behaviour. Looks like that's added to the textarea by Github. Natively tab goes from textarea to the next element even if there's an uncollapsed selection.

**Comment by ellatrix** [comment]
Related: #37336

**Comment by afercia** [comment]
> You can actually use space and backspace to indent and unindent with the caret on the first position of a list.

Yes and I'd like to note that also on Google Docs Tab / Shift+Tab indent and unindent only when:
- caret is at first position of a list item
- a list item 'row' is selected

When the caret is in the middle or at the end, Tab / Shift+Tab don't do anything, and for good reasons.

Additionally, in the WordPress Editor, Shift+Tab was established long time ago to be _the main affordance_ to navigate with the keyboard from the selected block to the block toolbar. This must always work under all circumstances and never break.

**Comment by annezazu** [comment]
After a review by core editor triage and core editor tech leads on the WordPress 6.5 release team, [this issue is being removed from the release. ](https://wordpress.slack.com/archives/C02QB2JS7/p1707153082189369?thread_ts=1707153067.570549&cid=C02QB2JS7)

**Comment by Marc-pi** [comment]
@ellatrix ok, done through https://github.com/WordPress/gutenberg/pull/59199
thanks ! close
